db/push.go

```diff
@@ -0,0 +1,96 @@
+package db
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	nomsjson "roci.dev/diff-server/util/noms/json"
+)
+
+// These request/response types might want to live in a different location, eg 
+// in diff-server/serve/types.
+type Mutation struct {
+	ID   uint64          `json:"id"`
+	Name string          `json:"name"`
+	Args json.RawMessage `json:"args"`
+}
+
+type BatchPushRequest struct {
+	ClientID  string     `json:"clientId"`
+	Mutations []Mutation `json:"mutations"`
+}
+
+type MutationInfo struct {
+	ID    string `json:"id"`
+	Error string `json:"error"`
+}
+
+type BatchPushResponse struct {
+	MutationInfos []MutationInfo `json:"mutationInfos"`
```

Comment:
*infos* is awkward. The docs call for just `mutations` as the field name. if you don't like that, what about `results`?

Reply:
results implies that it will be always be there, which it wont eg async endpoints. info says it is informational to me.